feat: ROOT-64: fully generated schema

robot-ci-heartex opened the pull request:
<!-- Follow Merge -->
This PR was automatically generated via Follow Merge. [Workflow Run](https://github.com/HumanSignal/label-studio/actions/runs/16657725968)
Please ensure that all linked upstream Pull Requests are merged before proceeding with this one.
- https://github.com/HumanSignal/label-studio-sdk/pull/490
<!-- Follow Merge -->

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/16304954070

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/16305448600

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/16305872017

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/16355688197

robot-ci-heartex committed:
Merge branch 'develop' into 'fb-ROOT-64'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/16355688197

jombooth committed:
some hacking around on decorators

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/16356659197

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/16357299304

robot-ci-heartex committed:
Merge branch 'develop' into 'fb-ROOT-64'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/16357299304

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/16358051081

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/16358553417

jombooth committed:
add paginated_response_schema and fix import API

jombooth committed:
Merge branch 'fb-ROOT-64' of github.com:HumanSignal/label-studio into fb-ROOT-64

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/16359633811

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/16359798935

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/16363608264

robot-ci-heartex committed:
Merge branch 'develop' into 'fb-ROOT-64'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/16363608264

jombooth committed:
type some SerializerMethodFields

jombooth committed:
Merge branch 'fb-ROOT-64' of github.com:HumanSignal/label-studio into fb-ROOT-64

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/16363846328

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/16363994614

jombooth committed:
remove unnecessary responses override

jombooth committed:
Merge branch 'fb-ROOT-64' of github.com:HumanSignal/label-studio into fb-ROOT-64

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/16378652006

robot-ci-heartex committed:
Merge branch 'develop' into 'fb-ROOT-64'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/16378652006

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/16379027348

jombooth committed:
fix download_sync call

jombooth committed:
Merge branch 'fb-ROOT-64' of github.com:HumanSignal/label-studio into fb-ROOT-64

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/16379493634

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/16379690657

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/16379886251

robot-ci-heartex committed:
Merge branch 'develop' into 'fb-ROOT-64'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/16379886251

jombooth committed:
fix predictions tests

jombooth committed:
Merge branch 'fb-ROOT-64' of github.com:HumanSignal/label-studio into fb-ROOT-64

jombooth committed:
merge develop

jombooth commented:
/fm sync
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/16380277929)

jombooth committed:
address bare string responses that will fail in the SDK & fix lockfile

jombooth commented:
/fm sync
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/16380821018)

jombooth committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/16380821018

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/16381051031

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/16381996916

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/16384574776

robot-ci-heartex committed:
Merge branch 'develop' into 'fb-ROOT-64'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/16384574776

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/16405850527

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/16406180067

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/16406407859

jombooth committed:
few remaining nits

jombooth committed:
Merge branch 'fb-ROOT-64' of github.com:HumanSignal/label-studio into fb-ROOT-64

jombooth committed:
nullify decorators for Enterprise

jombooth commented:
/fm sync
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/16406857899)

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/16406832618

jombooth committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/16406857899

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/16409562170

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/16410173584

robot-ci-heartex committed:
Merge branch 'develop' into 'fb-ROOT-64'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/16410173584

jombooth committed:
add x-fern-header to auth

jombooth committed:
Merge branch 'fb-ROOT-64' of github.com:HumanSignal/label-studio into fb-ROOT-64

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/16421761196

robot-ci-heartex committed:
Merge branch 'develop' into 'fb-ROOT-64'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/16421761196

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/16423454073

robot-ci-heartex committed:
Merge branch 'develop' into 'fb-ROOT-64'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/16423454073

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/16423722414

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/16423933180

robot-ci-heartex committed:
Merge branch 'develop' into 'fb-ROOT-64'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/16423933180

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/16426070014

robot-ci-heartex committed:
Merge branch 'develop' into 'fb-ROOT-64'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/16426070014

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/16426367051

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/16426795352

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/16427074228

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/16427474241

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/16429519326

robot-ci-heartex committed:
Merge branch 'develop' into 'fb-ROOT-64'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/16429519326

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/16488648328

robot-ci-heartex committed:
Merge branch 'develop' into 'fb-ROOT-64'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/16488648328

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/16488756610

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/16488881730

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/16489051699

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/16513944586

robot-ci-heartex committed:
Merge branch 'develop' into 'fb-ROOT-64'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/16513944586

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/16514278711

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/16514423404

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/16514480131

jombooth committed:
fix more param types

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/16530491130

robot-ci-heartex committed:
Merge branch 'develop' into 'fb-ROOT-64'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/16530491130

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/16530853862

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/16530917320

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/16531347836

jombooth commented:
/fm sync
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/16601142502)

jombooth committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/16601142502

jombooth committed:
Merge branch 'develop' into 'fb-ROOT-64'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/16601142502

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/16605484405

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/16605575976

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/16605980110

jombooth committed:
add convert API response

jombooth committed:
Merge branch 'fb-ROOT-64' of github.com:HumanSignal/label-studio into fb-ROOT-64

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/16606232861

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/16606365575

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/16607094988

robot-ci-heartex committed:
Merge branch 'develop' into 'fb-ROOT-64'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/16607094988

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/16635479524

robot-ci-heartex committed:
Merge branch 'develop' into 'fb-ROOT-64'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/16635479524

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/16635760188

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/16636093410

jombooth commented:
/git merge develop
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/16655490459)
> Successfully merged:  delete mode 100644 web/libs/editor/src/components/Timeline/Controls/VideoConfigControl.scss

jombooth committed:
Merge branch 'develop' into 'fb-ROOT-64'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/16655490459

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/16657725968

robot-ci-heartex committed:
Merge branch 'develop' into 'fb-ROOT-64'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/16657725968

hakan458 commented on label_studio/data_export/api.py:
```diff
@@ -154,8 +153,8 @@ def get(self, request, *args, **kwargs):
-            'x-fern-sdk-group-name': 'projects',
-            'x-fern-sdk-method-name': 'export',
+            'x-fern-sdk-group-name': ['projects', 'exports'],
+            'x-fern-sdk-method-name': 'download_sync',
```
why this name?

jombooth replied:
it's what we called easy export before: https://label-studio.docs.buildwithfern.com/api-reference/api-reference/projects/exports/download-sync and I tried to avoid changing the names of methods in the generated SDK

hakan458 commented on label_studio/data_manager/api.py:
```diff
@@ -259,6 +259,30 @@ def paginate_queryset(self, queryset, request, view=None):
             return self.paginate_totals_queryset(queryset, request, view)
         return self.sync_paginate_queryset(queryset, request, view)
 
+    def get_paginated_response_schema(self, schema):
```
what is the schema passed into here exactly?

jombooth replied:
It's used by Spectacular and then Fern to describe the pagination behavior at this endpoint (in particular, that the field containing the list of tasks is called `tasks` rather than `results`)

hakan458 commented on label_studio/jwt_auth/auth.py:
```diff
@@ -50,4 +50,9 @@ def get_security_definition(self, auto_schema):
+            'x-fern-header': {
+                'name': 'api_key',
+                'env': 'LABEL_STUDIO_API_KEY',
+                'prefix': 'Token ',
```
One thing I never figured out is... 
api_key param supports both legacy token, and jwt. The docs specify -H "Authorization: Token [your-token]" but it needs to be `Bearer` when using jwt. Confusing for users. Not sure if there is a clean way to show both 

https://humansignal.slack.com/archives/C072ZLBQ6SJ/p1749753981577629?thread_ts=1749750510.009489&cid=C072ZLBQ6SJ

jombooth replied:
I think we could figure this out, but I wanted to keep consistent with what we had before for this initial version

hakan458 commented on label_studio/jwt_auth/views.py:
```diff
@@ -47,6 +52,11 @@ class TokenExistsError(APIException):
+        extensions={
+            'x-fern-sdk-group-name': 'jwt_settings',
+            'x-fern-sdk-method-name': 'update',
+            'x-fern-audiences': ['public'],
```
when do we use non-public audiences? just curious. and is there a default if we dont specify?

jombooth replied:
We have a few endpoints that are marked `'internal'` and these don't show up in fern. In practice, this is the same as not applying the fern extensions at all.

hakan458 commented on label_studio/ml/api.py:
```diff
@@ -440,7 +440,25 @@ def post(self, request, *args, **kwargs):
+        responses={
+            200: OpenApiResponse(
+                description='List of available versions.',
+                response={
```
would it be cleaner to add a serializer here to avoid manual schema?

jombooth replied:
It would, but I wanted to minimize changes to view code; the way to really capture value from this would be to add a serializer and also use it in the view. I'll add a TODO for that

jombooth committed:
add TODO re: serializer

jombooth commented:
/git merge develop
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/16683757098)
> Successfully merged:  6 files changed, 26 insertions(+), 12 deletions(-)

jombooth committed:
Merge branch 'develop' into 'fb-ROOT-64'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/16683757098

matt-bernstein commented on label_studio/webhooks/api.py:
```diff
@@ -144,7 +144,27 @@ def get_queryset(self):
         summary='Get all webhook actions',
         description='Get descriptions of all available webhook actions to set up webhooks.',
-        responses={'200': 'Object with description data.'},
+        responses={
```
Same reason for not adding a serializer here as in `ml/api.py`?

jombooth replied:
merged before I saw this - actually this one's slightly different, in that it's not super easy (as far as I know) to programmatically generate a serializer. The response from this endpoint is a bit weird.

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio-enterprise/actions/runs/16684413709
